src/service/models.py

```diff
@@ -0,0 +1,45 @@
+"""
+Pydantic models for the Spark Manager API.
+"""
+
+import re
+
+from pydantic import BaseModel, Field, field_validator
+
+
+class SparkClusterConfig(BaseModel):
+    """Configuration for creating a new Spark cluster."""
+
+    worker_count: int = Field(default=2, description="Number of worker nodes")
+    worker_cores: int = Field(default=10, description="CPU cores per worker")
+    worker_memory: str = Field(
+        default="10G", description="Memory per worker (format: {int}G)"
+    )
+    master_cores: int = Field(default=10, description="CPU cores for master node")
+    master_memory: str = Field(
+        default="10G", description="Memory for master node (format: {int}G)"
+    )
```

Comment:
Are these incoming fields? If so they should have minimum and maximum limits

Reply:
Those are incoming fields - we currently don't know min/max for it.

Reply:
Min must be > 0 and some reasonable number < inf. I would put in reasonable placeholders for now

Reply:
👍